spec/ruby/core/io/set_encoding_by_bom_spec.rb

```diff
@@ -67,5 +67,11 @@
 
       -> { @io.set_encoding_by_bom }.should raise_error(ArgumentError, 'encoding is set to UTF-8 already')
     end
+
+    it 'returns exception if encoding conversion is already set' do
+      @io.set_encoding(nil, Encoding::UTF_8)
+
+      -> { @io.set_encoding_by_bom }.should raise_error(ArgumentError, 'encoding conversion is set')
+    end
```

Comment:
I'm not sure if we add to the specs here or the workflow is to add to https://github.com/ruby/spec first and then update here later, but I added this because this behaviour of the MRI implementation was not covered by the existing specs.

Reply:
It's fine to add them here, and they will be synchronized to ruby/spec.

Reply:
Thanks @eregon! I've sent an email requesting to be added to the contributors list on behalf of Shopify, but the PR should be good to review now.

Reply:
I added you the list, the OCA check should update the PR soon